### PR TITLE
Add Gift Card integration

### PR DESCRIPTION
**Pull request by gcatanese**
Add Gift Card integration example

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Get started by navigating to one of the supported demos below.
 |                                          Demos | Description                                              | Details                            |
 |-----------------------------------------------:|:---------------------------------------------------------|:-----------------------------------|
 |         [`Checkout Example`](checkout-example) | E-commerce checkout flow with different payment methods. | [See below](#checkout-example)     | 
+|        [`Gift Card Example`](giftcard-example) | Gift Cards checkout flow using partial orders.           | [See below](#giftcard-example)     |
```
Comment (Kwok-he-Chu) [inline README.md:14]: Link invalid: [See below(#giftcard-example)] should be: [See below(#gift-card-example)]

File: giftcard-example/README.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/adyen-examples/adyen-java-spring-online-payments/tree/main/giftcard-example)
+
+_NOTE: To allow the Adyen Drop-In and Components to load, you have to add `https://*.gitpod.io` as allowed origin for your chosen set of [API Credentials](https://ca-test.adyen.com/ca/ca/config/api_credentials_new.shtml)_
+
+## Details
+
+This repository includes examples of PCI-compliant UI integrations for online payments with Adyen. Within this demo app, you'll find a simplified version of an e-commerce website, complete with commented code to highlight key features and concepts of Adyen's API. Check out the underlying code to see how you can integrate Adyen to give your shoppers the option to pay with their preferred payment methods, all in a seamless checkout experience.
+
+![Card checkout demo](src/main/resources/static/images/cardgiftcard.gif)
```
Comment (Kwok-he-Chu) [inline giftcard-example/README.md:18]: Change: [Card checkout demo] to [Card Gift Card demo]

File: giftcard-example/README.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+## Supported Integrations
+
+Before testing, please make sure to [add the gift card payment method(s) to your Adyen Account](https://docs.adyen.com/payment-methods#add-payment-methods-to-your-account).
+The Demo leverages Adyen's API Library for Java ([GitHub](https://github.com/Adyen/adyen-java-api-library) | [Docs](https://docs.adyen.com/development-resources/libraries#java)).
```
Comment (Kwok-he-Chu) [inline giftcard-example/README.md:23]: Link https://docs.adyen.com/development-resources/libraries#java
should now be:
https://docs.adyen.com/development-resources/libraries?tab=java_2#java

File: giftcard-example/src/main/resources/templates/layout.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+
+    <link
+      rel="stylesheet"
+      href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
+      integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T"
+      crossorigin="anonymous"
+    />
+
+    <!-- Adyen JS from TEST environment (change to live for production)-->
```
Comment (Kwok-he-Chu) [inline giftcard-example/src/main/resources/templates/layout.html:23]: - [ ] Create an issue so we don't forget to update the E2E for gift cards and we can start migrating to the latest versions

**Review by Kwok-he-Chu — APPROVED** [review]
Approved, just minor text edits are left. Nicely done!

File: giftcard-example/src/main/resources/templates/giftcard.html [hunk 5]
```diff
@@ -0,0 +1,120 @@
+<html lang="en"
+      xmlns:th="http://www.thymeleaf.org"
+      xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
+      layout:decorate="~{layout}">
+<head>
+  <title>Checkout</title>
```
[the remark this answers is not in the document]
Comment (gcatanese) [reply giftcard-example/src/main/resources/templates/giftcard.html:6]: Gift Card Demo prefix is displayed already (added by Spring dynamically)

File: giftcard-example/src/main/resources/templates/dropin.html [hunk 6]
```diff
@@ -0,0 +1,33 @@
+<html lang="en"
+      xmlns:th="http://www.thymeleaf.org"
+      xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
+      layout:decorate="~{layout}">
+<head>
+  <title>Checkout</title>
```
[the remark this answers is not in the document]
Comment (gcatanese) [reply giftcard-example/src/main/resources/templates/dropin.html:6]: Gift Card Demo prefix is displayed already (added by Spring dynamically)

File: giftcard-example/src/main/resources/templates/layout.html (hunk 4 above)
Comment (gcatanese) [reply giftcard-example/src/main/resources/templates/layout.html:23]: See #82

**Pull request merged**